### PR TITLE
feat!: move some deps from direct `dependencies` to `peerDependencies` + drop `legacy-peer-deps=true`

### PR DESCRIPTION
**Pull request by nsbarsukov**
https://github.com/taiga-family/taiga-ui/blob/87eb95a7ba060f2589002c900cb98a75b6b590e7/.npmrc#L1

### CODE REVIEW

File: jest.config.ts [hunk 1]
```diff
@@ -34,6 +34,7 @@ const config: Config = {
     globals: {
         'ts-jest': {
             tsconfig: resolve(__dirname, 'tsconfig.spec.json'),
+            stringifyContentPathRegex: '\\.html$',
```
Comment (nsbarsukov) [inline jest.config.ts:37]: Learn more:
https://thymikee.github.io/jest-preset-angular/docs/guides/esm-support#manual-configuration

File: tsconfig.spec.json [hunk 2]
```diff
@@ -2,6 +2,7 @@
     "extends": "./tsconfig.json",
     "exclude": ["projects/demo-cypress", "**/scripts/**/*.ts"],
     "compilerOptions": {
+        "target": "es2016",
```
Comment (nsbarsukov) [inline tsconfig.spec.json:5]: Learn more:
* https://github.com/nrwl/nx/issues/15390#issuecomment-1457792176
* https://thymikee.github.io/jest-preset-angular/docs/getting-started/installation#configuration
Screenshot 2024-02-13 at 16 52 37

File: projects/testing/package.json [hunk 3]
```diff
@@ -2,14 +2,10 @@
...
     "description": "Utils functions for testing Taiga UI packages",
-    "dependencies": {
-        "cypress-image-snapshot": "4.0.1"
-    },
     "devDependencies": {
-        "@types/cypress-image-snapshot": "3.1.9",
         "@types/jest": "29.5.12",
         "jest": "29.7.0",
-        "jest-preset-angular": "12.2.6",
+        "jest-preset-angular": "13.1.5",
```
Comment (nsbarsukov) [inline projects/testing/package.json:8]: `12.2.6` is incompatible with `jest@29`

File: .github/renovate.json
Comment (nsbarsukov) [inline .github/renovate.json:1]: New changes inside this file are already tested inside `Maskito`:

https://github.com/taiga-family/maskito/blob/1b1b35e0dfd09ec83d8d5e210a01dce536aff159/.github/renovate.json#L4-L12

**Review by splincode — APPROVED** [review]

**Review by vladimirpotekhin — APPROVED** [review]

**Pull request merged**